workflow/workflow_commands.go

```diff
@@ -1445,6 +1448,76 @@ func TraceWorkflow(c *cli.Context) error {
 	return nil
 }
 
+func UpdateWorkflow(c *cli.Context) error {
+	wid := c.String(common.FlagWorkflowID)
+	rid := c.String(common.FlagRunID)
+	name := c.String(common.FlagName)
+	uid := c.String(common.FlagUpdateID)
+	first_execution_run_id := c.String(common.FlagUpdateFirstExecutionRunID)
+	args, err := common.UnmarshalInputsFromCLI(c)
+	if err != nil {
+		return err
+	}
+	waitPolicy, ok := updateWaitPolicyMap[c.String(common.FlagUpdateWaitPolicy)]
+	if !ok {
+		return fmt.Errorf("must specify valid wait policy: %v", strings.Join(mapKeysToArray(updateWaitPolicyMap), ", "))
+	}
+	if (len(name) == 0 && len(uid) == 0) || (len(name) > 0 && len(uid) > 0) {
+		return fmt.Errorf("either %v or %v should be provided", color.Yellow(c, "--%v", common.FlagName), color.Yellow(c, "--%v", common.FlagUpdateID))
+	}
+
+	if len(name) > 0 {
+		request := sdkclient.UpdateWorkflowWithOptionsRequest{
+			WorkflowID:          wid,
+			RunID:               rid,
+			UpdateName:          name,
+			Args:                args,
+			FirstExecutionRunID: first_execution_run_id,
+			// TODO: uncomment when waitPolicy is available
+			// waitPolicy:          waitPolicy,
+		}
+		return updateWorkflowHelper(c, &request, waitPolicy)
+	} else {
+		// TODO: implement this when GetWorkflowUpdateHandle is available in SDKClient
+		return fmt.Errorf("Getting update result is not yet implemented")
+	}
+
+}
+
+// TODO: remove waitPolicy after it is available in the request
+func updateWorkflowHelper(c *cli.Context, request *sdkclient.UpdateWorkflowWithOptionsRequest, waitPolicy interface{}) error {
+	ctx, cancel := common.NewContext(c)
+	defer cancel()
+	sdk, err := client.GetSDKClient(c)
+	if err != nil {
+		return err
+	}
+
+	workflowUpdateHandle, err := sdk.UpdateWorkflowWithOptions(ctx, request)
+
+	if err != nil {
+		return fmt.Errorf("unable to update workflow: %w", err)
+	}
+
+	switch waitPolicy {
+	case enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_COMPLETED:
+		var valuePtr interface{}
+		err = workflowUpdateHandle.Get(ctx, &valuePtr)
+		if err != nil {
+			return fmt.Errorf("unable to update workflow: %w", err)
+		}
+		result := map[string]interface{}{
+			"Name":     request.UpdateName,
+			"UpdateID": workflowUpdateHandle.UpdateID(),
+			"Result":   valuePtr,
+		}
+
+		fmt.Println(color.Green(c, "update workflow succeeded:"))
```

Comment:
Maybe only print this line if stdout is a tty. If someone is piping this output to another tool (e.g. `jq`) then a user would prefer to just have the json output and nothing pretty.

Reply:
In general we should have a quiet option I think and use normal logging, but I don't know how bad the CLI is at respecting stdout decisions from callers. What does running a workflow or query do? We should copy that.

Reply:
That's another very good point, I am not sure if we have standard practice for the output or not. Right now there are some commands that print out json and some that print out messages. There is a `--color` option which disables color, but I wonder if we need a `--json` that change the output format to json for all commands. Let me follow up on that.

Reply:
Yeah I think the issue is that there is no consistency in this specific file. Some subcommands just print out messages, some, like query, print out a serialized version of an object, others, like describe, prints out json. 
Another thing is for testing we need to print to `app.Writer` to be able to capture the output while testing without adding stdout capture, so need to `FPrintf` all these to `app.Writer`

Reply:
Fro now I removed this message from the output per @MichaelSnowden suggestion and we only print out update information if this is successful. I tried to use the same approach as query by stringifying the result, but that does not look very good: 
Json
```
{
  "Name": "fetch_and_add",
  "Result": 23,
  "UpdateID": "a6dfc688-8a2b-41b4-ab3f-259ac31110eb"
}
```
vs stringify: 
```
map{Name:fetch_and_add, Result:23, UpdateID:a6dfc688-8a2b-41b4-ab3f-259ac31110eb}
```

So I keep this as JSON and will start a conversation about the format of the outputs to make them all consistent.